### PR TITLE
AEA-622, AEA-640 Add more details on decision maker in docs

### LINKED ISSUE #1071: Docs issues
**Is your feature request related to a problem? Please describe.**
A few issues with docs

**Describe the solution you'd like**
- [x] explain how to disable consistency checks on aea cli
- [x] copy useful stuff from here: https://fetchai.github.io/oef-sdk-python/user/communication-protocols.html
- [x] add explanation of `--local` version to `init`.
- [x] point to https://docs.fetch.ai/aea/package-imports/ in quickstart
- [x] explain why own package management is used compared to python's default one. just some intro would be nice
- [x] explain local vs web based registry
- [x] explain only package with greatest version is activated
- [x] disambiguate "internal state"
- [x] explain callback design of handlers/behaviours
- [x] explain mix of configuration and code for modularity
- [x] explain `self.context.decision_maker_message_queue.put_nowait(tx_msg)`
- [x] problems persist mostly about onboarding: better to start by introducing the protocols & dialogues > main thing to learn
- [x] execution interruption: explain & configure (next release)
- [x] > describe dm: (can think of it like a wallet manager + “econ brain”); interface & implementation split; needs extension; Let developers mount their own decision maker by specifying it as a single python file in core of the project and then it gets mounted.; Implement a base interface class and then provide a common reference implementation.
- [x] in `config.md` it should be `loop_mode` not `agent_loop`
- [x] point to third party logos somewhere in `concepts` section to show we integrate with many different projects: community: ipfs, libp2p, ethereum, cosmos

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR extends docs as per open issues.

## Fixes

#1071 , #1276 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[MASTER]
```
Comment (DavidMinarsch) [inline .pylintrc:1]: @marcofavorito added pylint; over time we can enable more checks

File: aea/__version__.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 #
 # ------------------------------------------------------------------------------
 
-"""Specifies the version of the TAC package."""
+"""Specifies the version of the AEA package."""
```
Comment (DavidMinarsch) [inline aea/__version__.py:20]: awkward :D

File: aea/crypto/ethereum.py [hunk 3]
```diff
@@ -93,7 +93,9 @@ def load_private_key_from_path(cls, file_name) -> Account:
         path = Path(file_name)
         with open(path, "r") as key:
             data = key.read()
-            account = Account.from_key(data)
+            account = Account.from_key(  # pylint: disable=no-value-for-parameter
```
Comment (DavidMinarsch) [inline aea/crypto/ethereum.py:96]: The methods here can be used both as class and instance methods; but pylint does not detect that

File: aea/protocols/generator.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #
 # ------------------------------------------------------------------------------
 """This module contains the protocol generator."""
+# pylint: skip-file
```
Comment (DavidMinarsch) [inline aea/protocols/generator.py:20]: This file needs serious refactoring!

File: aea/runtime.py [hunk 5]
```diff
@@ -244,7 +244,7 @@ def _stop(self) -> None:
             return
 
         def set_task():
-            self._stopping_task = self._loop.create_task(self._stop_coro())
+            self._stopping_task = self._loop.create_task(self._stop_runtime())
```
Comment (DavidMinarsch) [inline aea/runtime.py:247]: @solarw I think you missed that substitution. Am I correct?

File: aea/test_tools/click_testing.py [hunk 6]
```diff
@@ -24,6 +24,7 @@
 CLIRunner.invoke, in the 'finally' clause. More precisely, before reading from
 the testing outstream, it checks whether it has been already closed.
 """
+# pylint: skip-file
```
Comment (DavidMinarsch) [inline aea/test_tools/click_testing.py:27]: Not our code; also opened an issue to remove it

File: docs/api/aea.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ This class implements an autonomous economic agent.
 ```python
  | __init__(identity: Identity, connections: List[Connection], wallet: Wallet, ledger_apis: LedgerApis, resources: Resources, loop: Optional[AbstractEventLoop] = None, timeout: float = 0.05, execution_timeout: float = 0, is_debug: bool = False, max_reactions: int = 20, decision_maker_handler_class: Type[
  |             DecisionMakerHandler
- |         ] = DefaultDecisionMakerHandler, skill_exception_policy: ExceptionPolicyEnum = ExceptionPolicyEnum.propagate, loop_mode: Optional[str] = None, **kwargs, ,) -> None
+ |         ] = DefaultDecisionMakerHandler, skill_exception_policy: ExceptionPolicyEnum = ExceptionPolicyEnum.propagate, loop_mode: Optional[str] = None, runtime_mode: Optional[str] = None, **kwargs, ,) -> None
```
Comment (DavidMinarsch) [inline docs/api/aea.md:21]: Missing `runtime_mode` in docstring!

File: aea/__version__.py (hunk 2 above)
Comment (marcofavorito) [reply aea/__version__.py:20]: oh no...

**Pull request merged**